app/components/Views/QRTabSwitcher/QRTabSwitcher.test.tsx

```
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import QRTabSwitcher, { Screens } from './QRTabSwitcher';
+import { useRoute } from '@react-navigation/native';
+import { strings } from '../../../../locales/i18n';
+
+jest.mock('@react-navigation/native', () => ({
+  useNavigation: () => ({
+    navigate: jest.fn(),
+    navigation: {},
+  }),
+  createNavigatorFactory: () => ({}),
+  useRoute: jest.fn(() => ({
+    params: {
+      onScanError: jest.fn(),
+      onScanSuccess: jest.fn(),
+      initialScreen: 0,
+    },
+  })),
+}));
+
+jest.mock('../../../util/test/configureStore', () => {
+  const configureMockStore = jest.requireActual('redux-mock-store').default;
+  return () => configureMockStore([])();
+});
+
+jest.mock('@react-navigation/compat', () => {
+  const actualNav = jest.requireActual('@react-navigation/compat');
+  return {
+    ...actualNav,
+    withNavigation: (obj: React.ReactNode) => obj,
+  };
+});
+
+jest.mock('../QRScanner', () => jest.fn(() => null));
+jest.mock('../../UI/ReceiveRequest', () => jest.fn(() => null));
+
+describe('QRTabSwitcher', () => {
+  it('renders QRScanner by default', () => {
+    const { getByText } = render(<QRTabSwitcher />);
+    expect(getByText(strings('qr_tab_switcher.scanner_tab'))).toBeTruthy();
+  });
+
+  it('does not render tabber when disableTabber is true', () => {
+    // @ts-expect-error useRoute doesn't have mockReturnValue
```

If doens't have, can we add it instead of ignoring the error?

---

I'd rather not go out of the scope for this ticket

---

It's in the scope, it's just tweaking your test mock.

---

useRoute was not implemented by this ticket, and the ts-expect-error was introduced as part of the `lint:tsc` step. I don't think that modifying useRoute/typing it should be part of this PR.

---

The issue being thrown is:
```
Property 'mockReturnValue' does not exist on type '<T extends RouteProp<ParamListBase, string>>() => T'.ts(2339)
```
I have casted this to jest.Mock to fix this ts-lint issue.